src/mmw/js/src/core/views.js

```diff
@@ -79,11 +80,25 @@ var HeaderView = Marionette.ItemView.extend({
     },
 
     templateHelpers: function() {
-        var self = this;
+        var self = this,
+            currentActive = self.appState.get('active_page'),
+            isSelectArea = currentActive === coreUtils.selectAreaPageTitle,
+            isProjects = currentActive === coreUtils.projectsPageTitle,
+            isAnalyze = currentActive === coreUtils.analyzePageTitle,
+            modelPackageNames = _.pluck(settings.get('model_packages'), 'display_name'),
+            isModel = _.contains(modelPackageNames, currentActive),
+            isCompare = currentActive === coreUtils.comparePageTitle,
+            wasAnalyzeVisible = self.appState.get('was_analyze_visible'),
+            wasModelVisible = self.appState.get('was_model_visible'),
+            wasCompareVisible = self.appState.get('was_compare_visible');
```

Comment:
It's possible to come back from Analyze to Select Area where Select Area would have `.active` but Analyze would still exist. So no, it is not the case that only first and last will ever have `.active`